### PR TITLE
Fix fullscreen reader for notch device

### PR DESCRIPTION
**Pull request by onlymash**
- Notch device support fullscreen 
- Make fullscreen switching smoother

### CODE REVIEW

**Comment by onlymash** [comment]
- Before
VID_20200216_222002_02

- After
VID_20200216_222002_03

**Comment by wopian** [comment]
Won't this cause text to be cut off for those that have "Crop Borders" enabled?

**Comment by arkon** [comment]
This could replace #2121 if it's updated.

I'd prefer for this to be configurable.

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt [hunk 1]
```diff
@@ -646,6 +653,14 @@ class ReaderActivity : BaseRxActivity<ReaderPresenter>() {
             }
         }
 
+        @TargetApi(Build.VERSION_CODES.P)
+        private fun setCutoutShort(enabled: Boolean) {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt]: Is there a way to force a re-layout after this? It seems that toggling the option doesn't make it take effect until I show/hide the toolbars again.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -184,6 +184,7 @@
 
       <!-- Reader section -->
     <string name="pref_fullscreen">Fullscreen</string>
+    <string name="pref_cutout_short">Content renders into the cutout area</string>
```
Comment (arkon) [inline app/src/main/res/values/strings.xml]: Maybe something shorter like "Show content in cutout area"?

File: app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsReaderController.kt [hunk 3]
```diff
@@ -74,6 +74,13 @@ class SettingsReaderController : SettingsController() {
             titleRes = R.string.pref_fullscreen
             defaultValue = true
         }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsReaderController.kt]: It'd be nice if this only showed up when there actually is display cutout(s), but I'm not sure if it's actually possible to check that.

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt (hunk 1 above)
Comment (onlymash) [reply app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt]: recreate activity(the worst practice)

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt]: Hmm... I guess it's fine for now then. We can revisit it if people complain.

**Pull request merged**